packages/ember-data/lib/system/store.js

```diff
@@ -1045,18 +1045,11 @@ Store = Service.extend({
   */
   all: function(modelName) {
     var typeClass = this.modelFor(modelName);
-    var typeMap = this.typeMapFor(typeClass);
-    var findAllCache = typeMap.findAllCache;
 
-    if (findAllCache) {
-      this.recordArrayManager.updateFilter(findAllCache, typeClass);
-      return findAllCache;
-    }
-
-    var array = this.recordArrayManager.createRecordArray(typeClass);
+    var liveRecordArray = this.recordArrayManager.liveRecordArrayFor(typeClass);
+    this.recordArrayManager.updateFilter(liveRecordArray, typeClass);
```

Comment:
> It seems like it will end up to a noop for liveRecordArrays

The name of `updateFilter` does not correctly reflect what's going on here: it calls `updateRecordArray` for each currently loaded record of type `typeClass`; and since there is a [hack](https://github.com/emberjs/data/blob/58c2cf65eb492165c56ba561803357b99020305a/packages/ember-data/lib/system/record-array-manager.js#L118) inside `updateRecordArray` at the moment, it also adds all records to the array. I will address in this PR ...